### PR TITLE
mute_alerts.py : Ability to mute health alerts

### PR DESCRIPTION
**Pull request by pdhiran**
These additions will allow us to test the new mute health alerts feature.
All the various CLI options are used to test the new feature introduced in 5.0

Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [x] Create a test case in Polarion reviewed and approved.
- [x] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [x] Update Polarin Test with Automation script details and update automation fields
- [x] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

**Comment by pdhiran** [comment]
@psathyan as per 
1. https://github.com/red-hat-storage/cephci/pull/395#discussion_r586115376, I have clubbed the below steps into another method. 
generated_health_alert
mute_health_alert

2. https://github.com/red-hat-storage/cephci/pull/395#discussion_r585676545, I have removed the non-configured daemons from the conf file.

**Comment by pdhiran** [comment]
pass logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614776347760/
email alert : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614776347760/result.html

File: suites/pacific/rados/sanity_rados.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                placement:
+                  label: mon
+          - config:
+              command: apply
+              service: osd
+              args:
+                all-available-devices: true
+      destroy-cluster: false
+      abort-on-fail: true
+  - test:
```
Comment (psathyan) [inline suites/pacific/rados/sanity_rados.yaml:51]: Can we remove this test for the time being? There is a patch set submitted by Sunil that would support retrieval of the cluster status. We could use that here.

Also verify is a misnomer here as we only get the status and no evaluation is done on the retrieved data.

File: tests/rados/mute_alerts.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    all_alerts = get_alerts(cephadm)
+    alert_list = ["MON_DISK_BIG", "OSDMAP_FLAGS"]
+    if all_alerts["active_alerts"]:
+        log.info(
+            f"There are health alerts generated on the cluster. Alerts : {all_alerts['active_alerts']}"
+        )
+    else:
+        log.info("Cluster Health is ok. \n Generating a alert to verify feature")
+
+    # Scenario 1 : Verify the auto-unmute of alert after TTL
```
Comment (psathyan) [inline tests/rados/mute_alerts.py:37]: I like the comments added to each section.

IMO, we should have separate functions executing the `scenario`. What do you think?

File: suites/pacific/rados/sanity_rados.yaml (hunk 1 above)
Comment (pdhiran) [reply suites/pacific/rados/sanity_rados.yaml]: Removed the test

File: tests/rados/mute_alerts.py (hunk 2 above)
Comment (pdhiran) [reply tests/rados/mute_alerts.py:37]: Added methods that can execute scenario on the alerts provided.

**Comment by pdhiran** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1614850054698/

File: tests/rados/mute_alerts.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+
+def unmute_health_alert(alert: str, node: CephAdmin) -> bool:
+    """
+    Un-Mutes the health alert on the cluster
+    Args:
+        alert: Name of the alert to be muted
+        node: node on which command should be executed
+
+    Returns: True -> pass, 1 -> failure
```
Comment (psathyan) [inline tests/rados/mute_alerts.py:279]: I believe it should be `False --> Failure`.. the rest looks good to be me.

File: tests/rados/mute_alerts.py [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+def generate_health_alert(alert: str, node: CephAdmin, **kwargs) -> bool:
+    """
+    Method to generate various health alerts
+    Args:
+        alert: name of the alert to be generated
+        node: name of the installer node
+        clear: Bool value which specifies if the given alert should be cleared
+        kwargs: any other params that need to be sent for a particular alert
+
+    Returns: 0 -> pass, 1 -> failure
```
Comment (psathyan) [inline tests/rados/mute_alerts.py:311]: Could you fix the `docstring`?

Comment (pdhiran) [reply tests/rados/mute_alerts.py]: done

File: tests/rados/mute_alerts.py (hunk 3 above)
Comment (pdhiran) [reply tests/rados/mute_alerts.py]: done

**Review by psathyan — APPROVED** [review]
Looks good to me

**Review by SrinivasaBharath — APPROVED** [review]
LGTM

**Pull request merged**